main.tf

```diff
@@ -1,13 +1,58 @@
-resource "random_integer" "example" {
+resource "kubernetes_service_account" "service_account" {
   count = module.this.enabled ? 1 : 0
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
+  metadata {
+    name        = var.deployment_name
+    namespace   = var.kubernetes_namespace
+    annotations = var.service_account_annotations
   }
 }
 
-locals {
-  example = format("%v %v", var.example, join("", random_integer.example[*].result))
-}
+resource "kubernetes_deployment" "tfc_cloud_agent" {
+  count = module.this.enabled ? 1 : 0
+
+  metadata {
+    name      = var.deployment_name
```

Comment:
```suggestion
    name      = coalesce(var.deployment_name, module.this.id)
```

Reply:
I need to default `var.deployment_name` to `null` for this to work, but we should also add a default of `tfc-agent` at the end of the `coalesce` in case none of the typical label vars are passed in.